src/AESecb.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+  @file AESecb.cpp
+  Implementation of ECB mode of operation for AES-128 algorithm
+*/
+#include "AESecb.hpp"
+#include <iostream>
+
+
+void printEncryptOutput(std::vector<unsigned char>& output) {
+  std::cout << std::endl;
+  std::cout << "Encrypt result";
+  for (int i = 0; i < output.size() / 16; i++){
+    std::cout << std::endl;
+    std::cout << "Block " << i << ": ";
+    for (int j = 0; j < 16; j++) {
+      std::cout << std::hex << (int) output[j+(i*16)];
+      std::cout << " ";
+    }
+  }
+}
+
+
+/**
+  Implements cipher with ECB mode of operation
+  @param input: array of hex values representing plaintext
+  @param output: array of hex values representing (padded) ciphertext
+  @param key: key to use
+  @param keysize: size of the key
+  @return array of decrypted values
+*/
+void encrypt_ecb(std::array<unsigned char, 16>& input, std::vector<unsigned char>& output, unsigned char* key, unsigned int keysize) {
```

Comment:
I think we could make input a const reference so it would be const std::array<unsigned char, 16>& input. This would just signify that input is not modified. Also we should make the key a vector as well.

Reply:
Why not make the key a const std::array as well?

Reply:
Key size is also variable, it could be 16, 24, or 32. Also I just noticed that input could be any size as well. That should also be a vector.

Reply:
Right. Defining NUM_BYTES as 16 had me fooled.